ili9341/ili9341.go

```diff
@@ -200,6 +205,13 @@ func (d *Device) DrawRGBBitmap8(x, y int16, data []uint8, w, h int16) error {
 	return nil
 }
 
+// DrawBitmap copies the bitmap to the internal buffer on the screen at the
+// given coordinates. It returns once the image data has been sent completely.
+func (d *Device) DrawBitmap(x, y int16, bitmap pixel.Image[pixel.RGB565BE]) error {
```

Comment:
A suggestion based on personal preference when working with generics:

Generic data structures can get tedious to write out and decrease legibility. I've found that using type aliases to declare shorthand common generic types is quite idiomatic:
```go
type Image = pixel.Image[pixel.RGB565BE]
```
You then use the `Image` type anywhere you'd use `pixel.Image[pixel.RGB565BE]`. A caveat (or benefit even) is that users will see the `Image` type if you export it, so you can even provide the convenience `ili9341.Image` type along with your device library so that generics don't pollute user facing APIs.

The greatest benefit of type aliases for generics I think comes from not having to add type parameters to structures which contain the type as one of the fields, thus preventing viral spreading of type parameters in a codebase.

Reply:
I guess I could add it, if needed. Though I don't see the need for it? I haven't needed anything like this for the three small projects I used the `pixel.Image` API with.

> The greatest benefit of type aliases for generics I think comes from not having to add type parameters to structures which contain the type as one of the fields, thus preventing viral spreading of type parameters in a codebase.

Yes, generics can get quite viral. See for example my tinygl package, which has the type parameter on almost _every_ type/struct/method: https://pkg.go.dev/github.com/aykevl/tinygl
However, I don't really see a way around this. For a specific application (where you are fine with hardcoding the pixel format) you can certainly make a type alias like this. But when you do that, you lose the ability to switch the pixel format. (For example: for my [watch](https://github.com/aykevl/things/tree/master/watch) I use RGB444 instead of RGB565 for improved speed, and the simulator uses RGB888).

So in short, I could add `ili9341.Image`, but I'm not sure it's really necessary/useful. And users can define it themselves if they really want to.

Reply:
Yes! agreed 100%, including the last EDIT note! :orange_heart:.